src/networking.c

```diff
@@ -41,11 +41,40 @@
 #include <math.h>
 #include <ctype.h>
 #include <stdatomic.h>
+#include "intset.h"
+#include <stdbool.h>
+
+/**
+ * This struct is used to encapsulate filtering criteria for operations on clients
+ * such as identifying specific clients to kill or retrieve. Each field in the struct
+ * represents a filter that can be applied based on specific attributes of a client.
+ */
+typedef struct {
+    /** A set of client IDs to filter. If NULL, no ID filtering is applied. */
+    intset *ids;
+    /** Maximum age (in seconds) of a client connection for filtering.
+     * Connections younger than this value will not match.
+     * A value of 0 means no age filtering. */
```

Comment:
Ah, I'm wrong, sorry. I understand "younger than this value will not match" means they will not be killed. The docs for CLIENT KILL says "Closes all the connections that are older than the specified age, in seconds".

So the value is the max age for clients to *not match*, which makes sense for CLIENT KILL but maybe very weird for CLIENT LIST?

Do we want `CLIENT LIST MAXAGE 1000` to mean "list all clients older than 1000"? It sounds illogical, but OTOH, I guess people will use CLIENT LIST first to list some clients and then CLIENT KILL with the same filters to kill them, so for this use case, the logic needs to be the same. :thinking:

Can we keep this behavior but come up with a good formulation for documenting this in CLIENT LIST?

Reply:
The safest solution is maybe to skip MAXAGE for CLIENT LIST. To do that, we can add a flag like `allow_maxage` for `parseClientFilters`. When `allow_maxage` is false (for CLIENT LIST), it will be a syntax error. Makes sense?

Reply:
> Do we want CLIENT LIST MAXAGE 1000 to mean "list all clients older than 1000"? It sounds illogical

Yeah, honestly it was a bit strange to me as well because KILL and LIST are kinda opposite in what they do. But currently, we return the list of clients with age greater than the specified maxage.

> The safest solution is maybe to skip MAXAGE for CLIENT LIST

Yes we can always skip it. But would we want to support this in the future? If yes, maybe we can properly document the behavior and keep supporting the maxage filter.

@madolson @hpatro would love to get your thoughts on this as well.